src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb

```diff
@@ -1162,9 +1162,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                             emitExtensionAttribute = ThreeState.True
                         End If
                     End If
-
                 End If
 
+                Debug.Assert(_lazyEmitExtensionAttribute = ThreeState.Unknown OrElse
+                             _lazyEmitExtensionAttribute = emitExtensionAttribute)
```

Comment:
order is unimportant ( I could use Or here) and assert would be racy even if we read once since a bad value could be installed right after we checked.

I think this is as good as we can have. "Assert we have no races here" is typically at best a best effort. 
The old assert was also racy and eventually caught something.